### ISSUE TITLE
Doesn't support dasherized keys

### ISSUE THREAD

**Issue by averydev** [b2beauty/jsonapi-utils#22]
The JSONAPI spec recommends using dasherized keys
http://jsonapi.org/recommendations/#naming

However, that breaks with jsonapi-utils.

  jsonapi-resources 0.7
  jsonapi-utils (~> 0.4.5)

With a request such as this:
```
{
  "data": {
    "id": "58",
    "type": "doctor-family-members",
    "links": {
      "self": "http://localhost:3000/doctor-family-members/58"
    },
    "attributes": {
      "doctor-id": 14,
      "family-member-id": 5,
      "min-visits": 2000,
      "avg-extra-visits": 3000,
      "doctor-name": "MATTHEW PARSONS"
    },
    "relationships": {
      "doctor": {
        "links": {
          "self": "http://localhost:3000/doctor-family-members/58/relationships/doctor",
          "related": "http://localhost:3000/doctor-family-members/58/doctor"
        }
      },
      "family-member": {
        "links": {
          "self": "http://localhost:3000/doctor-family-members/58/relationships/family-member",
          "related": "http://localhost:3000/doctor-family-members/58/family-member"
        }
      }
    }
  }
}
```

a param method of: 
`params.require(:data).require(:attributes).permit(:min_visits, :avg_extra_visits)`

Shows `min-visits` and `avg-extra-visits` as not permitted.

**Comment by tiagopog** [comment]
Thanks for sharing, @averydev. I'll check what's happening and patch a minor release with the fix as soon as I'm done with my current work.

Uh... is it related to #21?

**Comment by averydev** [comment]
Hi there @tiagopog fantastic!  I don't believe so, though it's possible.  I wasn't able to get around the dasherized problem to find out. I removed the extraneous item from the request to get that far :)

**Comment by tiagopog** [comment]
All right, I may come back later with some news. Thanks!

**Comment by tiagopog** [comment]
Here's the GOTCHA: using dasherized keys won't actually work with the default way of permitting params in Rails' ActionController. In order to avoid you getting stuck with this issue, for now you can access your permitted params by calling `@request.operations.first.data[:attributes]`. Anyway, I may write a better support for this feature – with some helper methods – on this weekend. Hope it helps you :-)

**Comment by tiagopog** [comment]
Closed in #23
